packages/itwinui-react/src/core/FileUpload/FileUploadCard.tsx

```diff
@@ -0,0 +1,126 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import { SvgUpload } from '../utils';
+import { FileUploadCardAction } from './FileUploadCardAction';
+import cx from 'classnames';
+import { FileUploadCardText } from './FileUploadCardText';
+import { FileUploadCardLabel } from './FileUploadCardLabel';
+import { FileUploadCardDescription } from './FileUploadCardDescription';
+import { FileUploadCardInput } from './FileUploadCardInput';
+import { FileUploadCardIcon } from './FileUploadCardIcon';
+
+export type FileUploadCardProps = {
+  /**
+   * List of files to pass (only needed when passing a custom action)
+   */
+  files?: File[];
```

Comment:
Maybe it should be `File` (singular) as this component is for one file.

Reply:
In the future, would we add another component or parameter for multiple files? If we add onto this component for multiple file support later on, `files` would be better long term.

Reply:
For multiple files we could do something like this:
```js
<FileUploadCardContainer>
  <Card1 />
  <Card2 />
</FileUploadCardContainer>
```
If we want to keep same component for both single and multiple files, we could add a flag "multiple" (or something like that) and add to prop type `files?: File | File[]` based on that flag. 

In both cases we should fix CardLabel and CardDescription components as they are using file[0] to get data.

Reply:
CardLabel and CardDescription have been changed

`files` has been changed to `data` so it works in the singular and plural sense which leaves room for a `multiple` flag that changes the prop type